File: proyectito_backend/ecommerce-minerva/java/com/proyecto/service/UserService.java
package com.proyecto.service;

import java.util.List;

import com.proyecto.Exception.UserException;
import com.proyecto.model.Usuario;
import com.proyecto.request.AdminCreateUserRequest;
import com.proyecto.request.AdminUpdateUserRequest;
import com.proyecto.request.ChangePasswordRequest;
import com.proyecto.request.UpdateProfileRequest;

public interface UserService {

	public Usuario findUserProfileByJwt(String jwt) throws UserException;

	void changePassword(Long userId, ChangePasswordRequest request) throws UserException;

	void updateUserProfile(Long userId, UpdateProfileRequest request) throws UserException;

	public List<Usuario> getAllUsers();

	public void toggleUserStatus(Long userId) throws UserException;

	public void deleteUser(Long userId) throws UserException;

	public Usuario createUserByAdmin(AdminCreateUserRequest req) throws UserException;
	public Usuario updateUserByAdmin(Long userId, AdminUpdateUserRequest req) throws UserException;
}

File: frontend_proyectito/ecommerce-ripley/src/app/services/user.service.ts
import { inject, Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import { environment } from '../../environments/environment';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models/api-response';
import { UsuarioResponse, AdminUsuarioResponse, AdminCreateUserRequest, AdminUpdateUserRequest } from '../models/user.model';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/api/users`;
  private adminApiUrl = `${environment.apiUrl}/admin/gestion-usuarios`;

  getProfile(): Observable<ApiResponse<UsuarioResponse>> {
    return this.http.get<ApiResponse<UsuarioResponse>>(`${this.apiUrl}/profile`);
  }

  // Admin Methods
  getAllUsers(): Observable<ApiResponse<AdminUsuarioResponse[]>> {
    return this.http.get<ApiResponse<AdminUsuarioResponse[]>>(`${this.adminApiUrl}/all`);
  }

  createUserByAdmin(req: AdminCreateUserRequest): Observable<ApiResponse<AdminUsuarioResponse>> {
    return this.http.post<ApiResponse<AdminUsuarioResponse>>(`${this.adminApiUrl}/create`, req);
  }

  updateUserByAdmin(userId: number, req: AdminUpdateUserRequest): Observable<ApiResponse<AdminUsuarioResponse>> {
    return this.http.put<ApiResponse<AdminUsuarioResponse>>(`${this.adminApiUrl}/${userId}`, req);
  }

  deleteUser(userId: number): Observable<ApiResponse<string>> {
    return this.http.delete<ApiResponse<string>>(`${this.adminApiUrl}/${userId}`);
  }

  toggleUserStatus(userId: number): Observable<ApiResponse<string>> {
    return this.http.put<ApiResponse<string>>(`${this.adminApiUrl}/${userId}/status`, {});
  }
}
